test/OpenFeature.Tests/OpenFeatureEventTests.cs

```diff
@@ -0,0 +1,413 @@
+                        payload => payload.ProviderName == testProvider.GetMetadata().Name && payload.Type == ProviderEventTypes.ProviderError
+                    ));
+        }
+
+        [Fact]
+        [Specification("5.1.2", "When a `provider` signals the occurrence of a particular `event`, the associated `client` and `API` event handlers MUST run.")]
+        [Specification("5.2.2", "The `API` MUST provide a function for associating `handler functions` with a particular `provider event type`.")]
+        [Specification("5.2.3", "The `event details` MUST contain the `provider name` associated with the event.")]
+        [Specification("5.2.4", "The `handler function` MUST accept a `event details` parameter.")]
+        [Specification("5.3.3", "Handlers attached after the provider is already in the associated state, MUST run immediately.")]
+        public async Task API_Level_Event_Handlers_Should_Be_Informed_About_Ready_State_After_Registering_Provider_Stale()
+        {
+            var eventHandler = Substitute.For<EventHandlerDelegate>();
+
+            var testProvider = new TestProvider();
+            await Api.Instance.SetProvider(testProvider);
+
+            testProvider.SetStatus(ProviderStatus.Stale);
+
+            Api.Instance.AddHandler(ProviderEventTypes.ProviderStale, eventHandler);
+
+            Thread.Sleep(1000);
+            eventHandler
+                .Received()
+                .Invoke(
+                    Arg.Is<ProviderEventPayload>(
+                        payload => payload.ProviderName == testProvider.GetMetadata().Name && payload.Type == ProviderEventTypes.ProviderStale
+                    ));
+        }
+
+        [Fact]
+        [Specification("5.1.2", "When a `provider` signals the occurrence of a particular `event`, the associated `client` and `API` event handlers MUST run.")]
+        [Specification("5.2.2", "The `API` MUST provide a function for associating `handler functions` with a particular `provider event type`.")]
+        [Specification("5.2.3", "The `event details` MUST contain the `provider name` associated with the event.")]
+        [Specification("5.2.4", "The `handler function` MUST accept a `event details` parameter.")]
+        [Specification("5.2.6", "Event handlers MUST persist across `provider` changes.")]
+        public async Task API_Level_Event_Handlers_Should_Be_Exchangeable()
+        {
+            var eventHandler = Substitute.For<EventHandlerDelegate>();
+
+            Api.Instance.AddHandler(ProviderEventTypes.ProviderReady, eventHandler);
+
+            var testProvider = new TestProvider();
+            await Api.Instance.SetProvider(testProvider);
+
+            var newTestProvider = new TestProvider();
+            await Api.Instance.SetProvider(newTestProvider);
+
+            Thread.Sleep(1000);
+            eventHandler.Received(2).Invoke(Arg.Is<ProviderEventPayload>(payload => payload.ProviderName == testProvider.GetMetadata().Name));
+        }
```

Comment:
Just because it's a different code-path, could you also assert this works as expected with `ProviderConfigurationChanged` events (manually emitted by the provider vs automatically by the SDK).